app/grandchallenge/retina_api/views.py

```diff
@@ -976,9 +978,38 @@ class ImageTextAnnotationViewSet(viewsets.ModelViewSet):
 
 
 class LandmarkAnnotationSetViewSet(viewsets.ModelViewSet):
-    permission_classes = (RetinaOwnerAPIPermission,)
+    permission_classes = (RetinaAPIPermission,)
     authentication_classes = (authentication.TokenAuthentication,)
     serializer_class = LandmarkAnnotationSetSerializer
     filter_backends = (filters.ObjectPermissionsFilter,)
     pagination_class = None
-    queryset = LandmarkAnnotationSet.objects.all()
+
+    def get_queryset(self):
+        """
+        Only return objects that belong to current user, except for retina_admins
+        :return: queryset
+        """
+        if is_in_retina_admins_group(self.request.user):
+            return LandmarkAnnotationSet.objects.all()
+        return self.request.user.landmarkannotationset_set.all()
+
+    @action(
+        detail=False,
+        methods=["get"],
+        url_path="for-image",
+        url_name="for-image",
+    )
+    def for_image(self, request):
+        """
+        Custom action for retrieving all LandmarkAnnotationSets that are associated
+        with a specific image defined by the query parameter "image_id"
+        :param request: Request
+        :return: Response
+        """
+        image_id = self.request.query_params.get("image_id")
+        image = get_object_or_404(Image.objects.all(), pk=image_id)
+        queryset = self.filter_queryset(self.get_queryset()).filter(
+            singlelandmarkannotation__image=image
+        )
+        serializer = self.get_serializer(queryset, many=True)
+        return Response(serializer.data)
```

Comment:
Should this be done using a filter on the list view? https://www.django-rest-framework.org/api-guide/filtering/

Reply:
Good point. I moved the functionality to the list view.